database/localdb/localdb.go

```diff
@@ -0,0 +1,59 @@
+package localdb
+
+import (
+	"chain/errors"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/tecbot/gorocksdb"
+)
+
+// DB provides access to a kv store.
+type DB struct {
+	store  *gorocksdb.DB
+	closed bool
+}
+
+// TODO(tessr): use 'Exec' instead of Put
+func (db *DB) Put(key string, value proto.Message) error {
+	encodedValue, err := proto.Marshal(value)
+	if err != nil {
+		return errors.Wrap(err)
+	}
+
+	// TODO(tessr): tune rocksdb. assess write options
+	wo := gorocksdb.NewDefaultWriteOptions()
+	return db.store.Put(wo, []byte(key), encodedValue)
+}
+
+func (db *DB) Get(key string, v proto.Message) error {
+	// TODO(tessr): tune rocksdb. assess read options
+	ro := gorocksdb.NewDefaultReadOptions()
+	slice, err := db.store.Get(ro, []byte(key))
+	defer slice.Free()
```

Comment:
I think we might want this defer after the err check.

It looks like Get will return a nil slice if err is non-nil: https://github.com/tecbot/gorocksdb/blob/master/db.go#L225-L228
And Slice.Free doesn't handle the case where s is nil: https://github.com/tecbot/gorocksdb/blob/master/slice.go#L30-L35